cuda2gcn/src/convert.cl

```diff
@@ -0,0 +1,167 @@
+/*===--------------------------------------------------------------------------
+ *                   ROCm Device Libraries
+ *
+ * This file is distributed under the University of Illinois Open Source
+ * License. See LICENSE.TXT for details.
+ *===------------------------------------------------------------------------*/
+
+#define ATTR __attribute__((always_inline))
+
+#define CONVERTM(A,B,m,n) ATTR B __nv_##A##2##B##_##m(A x) \
+    { return convert_##B##_##n(x); }
+
+#define CONVERT(A,B) \
+    CONVERTM(A, B, rd, rtn) \
+    CONVERTM(A, B, rn, rte) \
+    CONVERTM(A, B, ru, rtp) \
+    CONVERTM(A, B, rz, rtz)
+
+//-------- T __nv_double2float_rd
+//-------- T __nv_double2float_rn
+//-------- T __nv_double2float_ru
+//-------- T __nv_double2float_rz
+CONVERT(double, float)
+
+//-------- T __nv_double2int_rd
+//-------- T __nv_double2int_rn
+//-------- T __nv_double2int_ru
+//-------- T __nv_double2int_rz
+CONVERT(double, int)
+
+//-------- T __nv_float2int_rd
+//-------- T __nv_float2int_rn
+//-------- T __nv_float2int_ru
+//-------- T __nv_float2int_rz
+CONVERT(float, int)
+
+//-------- T __nv_int2float_rd
+//-------- T __nv_int2float_rn
+//-------- T __nv_int2float_ru
+//-------- T __nv_int2float_rz
+CONVERT(int, float)
+
+#define CONVERT2LLM(A,B,m,n) ATTR long __nv_##A##2ll_##m(A x) \
+    { return convert_long_##n(x); }
+
+#define CONVERT2LL(A) \
+    CONVERT2LLM(A, long, rd, rtn) \
+    CONVERT2LLM(A, long, rn, rte) \
+    CONVERT2LLM(A, long, ru, rtp) \
+    CONVERT2LLM(A, long, rz, rtz)
+
+//-------- T __nv_double2ll_rd
+//-------- T __nv_double2ll_rn
+//-------- T __nv_double2ll_ru
+//-------- T __nv_double2ll_rz
+CONVERT2LL(double)
+
+//-------- T __nv_float2ll_rd
+//-------- T __nv_float2ll_rn
+//-------- T __nv_float2ll_ru
+//-------- T __nv_float2ll_rz
+CONVERT2LL(float)
+
+#define CONVERT2ULLM(A,B,m,n) ATTR unsigned long __nv_##A##2ull_##m(A x) \
+    { return convert_ulong_##n(x); }
+
+#define CONVERT2ULL(A) \
+    CONVERT2ULLM(A, unsigned long, rd, rtn) \
```

Comment:
Use ulong instead of unsigned long and uint instead of unsigned int.  This may allow the use of fewer macros.